keps/0020-manual-plan-execution.md

```diff
@@ -173,8 +173,55 @@ Does not look like people are really doing this that much (they usually rather f
 
 #### As a stateful service operator (person) I want to be able to run backup on my stateful service when a backup plan is defined.
 
-### Implementation Details/Notes/Constraints [optional]
+### Implementation Details/Notes/Constraints
 
-TODO: choose a solution
+We've decided to go with `Have a separate CRD just for maintaining commands to execute a plan` solution.
 
-### Risks and Mitigations
\ No newline at end of file
+#### Different or unified ways to trigger a plan
+
+Right now, triggering plan executions is fully encapsulated in the instance_controller. It watches updates on Instance objects and decide if it needs to run plan and more importantly which plan (do we run upgrade plan or deploy plan for example?). With manually executed plans, we'll theoretically have two ways to trigger plan:
+
+1. *Non-explicit executions* that are triggered by changing the state of instance (either updating parameter or updating OperatorVersion referencing it - upgrade)
+1. *Explicit executions* (I as operator want to run plan - e.g. backup - right now)
```

Comment:
```suggestion
1. *Explicit executions* (I as operator want to run plan - e.g. backup - right now). Note, that 2. can be implemented through 1. by using plan parameter trigger like:
```
  - name: trigger_backup
    description: increment to trigger a backup plan
    default: 0
    trigger: "backup"
```
```